presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -85,4 +107,122 @@ public static long getFileSize(Page statisticsPage, int position)
         }
         return Optional.of(manifestBuilder.build());
     }
+
+    public static Map<String, String> updatePartitionMetadataWithFileNamesAndSizes(PartitionUpdate partitionUpdate, Map<String, String> metadata)
+    {
+        ImmutableMap.Builder<String, String> partitionMetadata = ImmutableMap.builder();
+        List<FileWriteInfo> fileWriteInfos = new ArrayList<>(partitionUpdate.getFileWriteInfos());
+
+        // Sort the file infos based on fileName
+        fileWriteInfos.sort(Comparator.comparing(info -> Integer.valueOf(info.getWriteFileName())));
+
+        // Compress the file names into a consolidated string
+        String fileNames = compressFileNames(fileWriteInfos.stream().map(FileWriteInfo::getWriteFileName).collect(toImmutableList()));
+
+        // Compress the file sizes
+        String fileSizes = compressFileSizes(fileWriteInfos.stream().map(FileWriteInfo::getFileSize).map(Optional::get).collect(toImmutableList()));
+
+        partitionMetadata.put(FILE_NAMES, fileNames);
+        partitionMetadata.put(FILE_SIZES, fileSizes);
+        partitionMetadata.put(MANIFEST_VERSION, VERSION_1);
+        partitionMetadata.putAll(metadata);
+
+        return partitionMetadata.build();
+    }
+
+    static String compressFileNames(List<String> fileNames)
+    {
+        if (fileNames.size() == 1) {
+            return fileNames.get(0);
+        }
+
+        boolean isContinuousSequence = true;
+        int start = 0;
+        for (String name : fileNames) {
+            if (start != Integer.parseInt(name)) {
+                isContinuousSequence = false;
+                break;
+            }
+            start++;
+        }
+
+        if (isContinuousSequence) {
+            return fileNames.get(fileNames.size() - 1);
+        }
+
+        return compressFileNamesUsingRoaringBitmap(fileNames);
```

Comment:
I think can blindly compressing with roaring bitmap. It compress continuous range very well.

Reply:
Not really. For a list of continuous 100 numbers, RoaringBitMap is taking 218 bytes. But where as "100" takes 3 bytes. So the current code is more memory efficient.